sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -63,18 +83,29 @@ class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
     }
   }
 
+  /**
+   * Dispatch events to registered StreamingQueryListeners. Only the events associated queries
+   * started in the same SparkSession as this ListenerBus will be dispatched to the listeners.
+   */
   override protected def doPostEvent(
       listener: StreamingQueryListener,
       event: StreamingQueryListener.Event): Unit = {
+    val runIdsToReportTo = activeQueryRunIds.synchronized { activeQueryRunIds.toSet }
```

Comment:
otherwise the code would look like 
```
case queryStarted: QueryStartedEvent =>
        if (activeQueryRunIds.synchronized { activeQueryRunIds.contains(queryStarted.runId) }) {
          listener.onQueryStarted(queryStarted)
        }
      case queryProgress: QueryProgressEvent =>
        if (activeQueryRunIds.synchronized { activeQueryRunIds.contains(queryProgress.progress.runId) }) {
          listener.onQueryProgress(queryProgress)
        }
      case queryTerminated: QueryTerminatedEvent =>
        if (activeQueryRunIds.synchronized { activeQueryRunIds.contains(queryTerminated.runId) }) {
          listener.onQueryTerminated(queryTerminated)
          activeQueryRunIds.synchronized { activeQueryRunIds -= queryTerminated.runId }
        }

```